.github/workflows/firebase-hosting-pull-request.yml

```diff
@@ -1,13 +1,17 @@
-name: Deploy to Firebase Hosting on PR
+name: Deploy Wolt Examples on PR
 'on': pull_request_target
 jobs:
   build_and_preview:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
         with:
-          channel: 'stable'
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - uses: subosito/flutter-action@v2.16.0
```

Comment:
@AcarFurkan can you please add here the line to read the flutter sdk version from the pubspec yaml. See example here: https://github.com/woltapp/wolt_modal_sheet/pull/192/files#diff-3ea54af4839eb75404d71b28252bead7e7ec8f676b1f815e1cde02629a75c165R22